### PR TITLE
Running our pulumi/pulumi examples on Windows

### PR DESCRIPTION
**Pull request by stack72**
Fixes: https://github.com/pulumi/home/issues/583

To-Do:

* [x] Go SDK Build / package
* [x] NodeJS SDK Build / Package
* [x] Python sdk build / package
* [x] Dotnet SDK build / package
* [x] Run Examples
* [x] Run Integration tests

### CODE REVIEW

**Comment by stack72** [comment]
There are a number of tests that don't run on windows as of today 1/26/20. In order to get some level of coverage on Windows, we are going to skip these tests and follow up with them afterwards. The tests that have been excluded on windows are:

```
* TestConfigCaptureNodeJS
* TestStackReferenceNodeJS
* TestStackReferencePython
* TestStackReferenceDotnet
* TestStackReferenceSecretsNodejs
* TestStackReferenceSecretsDotnet
* TestDynamicPython
* TestPartialValuesPython
```

File: pkg/testing/integration/program.go [hunk 1]
```diff
@@ -1673,7 +1672,7 @@ func (pt *programTester) prepareDotNetProject(projinfo *engine.Projinfo) error {
 			return errors.Wrap(err, "failed to find a local Pulumi NuGet package")
 		}
 		if len(matches) != 1 {
-			return errors.New(fmt.Sprintf("attempting to find a local Pulumi NuGet package yielded %v results", matches))
+			return errors.New(fmt.Sprintf("attempting to find a local Pulumi NuGet package yielded %d results", len(matches)))
```
Comment (lukehoban) [inline pkg/testing/integration/program.go]: Was this change needed?  It’s useful to know what the matches are when you hit this.

File: tests/integration/integration_test.go [hunk 2]
```diff
@@ -79,9 +78,8 @@ func TestEmptyPython(t *testing.T) {
 // TestEmptyGo simply tests that we can build and run an empty Go project.
 func TestEmptyGo(t *testing.T) {
 	integration.ProgramTest(t, &integration.ProgramTestOptions{
-		Dir:      filepath.Join("empty", "go"),
-		Quick:    true,
-		RunBuild: true,
+		Dir:   filepath.Join("empty", "go"),
+		Quick: true,
```
Comment (lukehoban) [inline tests/integration/integration_test.go]: Was the change to remove RunBuild required?

File: tests/integration/integration_test.go [hunk 3]
```diff
@@ -1253,6 +1265,9 @@ func TestResourceWithSecretSerialization(t *testing.T) {
 }
 
 func TestStackReferenceSecretsNodejs(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Temporarily skipping test on Windows")
```
Comment (lukehoban) [inline tests/integration/integration_test.go]: Minor - but could reference the issue number here.

File: pkg/testing/integration/program.go (hunk 1 above)
Comment (stack72) [reply pkg/testing/integration/program.go]: It wasn't - this was part of debugging - removed now

File: tests/integration/integration_test.go (hunk 3 above)
Comment (stack72) [reply tests/integration/integration_test.go]: done!

File: tests/integration/integration_test.go (hunk 2 above)
Comment (stack72) [reply tests/integration/integration_test.go]: I am going to retest this one before I merge this

Comment (stack72) [reply tests/integration/integration_test.go]: yeah this doesn't actually work on Windows for some reason:

```        [                       empty\go ] Updating (p-it-appveyor-v-go-119bcb5f):
         [                       empty\go ] @ Updating....
         [                       empty\go ] 
         [                       empty\go ]     pulumi:pulumi:Stack emptygo-p-it-appveyor-v-go-119bcb5f  error: an unhandled error occurred: problem executing program (could not run language executor): exec: "C:\\Users\\appveyor\\go\\bin\\emptygo": file does not exist
         [                       empty\go ]     pulumi:pulumi:Stack emptygo-p-it-appveyor-v-go-119bcb5f **failed** 1 error
         [                       empty\go ]  
         [                       empty\go ] Diagnostics:
         [                       empty\go ]   pulumi:pulumi:Stack (emptygo-p-it-appveyor-v-go-119bcb5f):
     1>[                       empty\go ]     error : an unhandled error occurred: problem executing program (could not run language executor): exec: "C:\\Users\\appveyor\\go\\bin\\emptygo": file does not exist [C:\Users\appveyor\go\src\github.com\pulumi\pulumi\build.proj]
         [                       empty\go ]  
         [                       empty\go ] Permalink: https://app.pulumi-staging.io/pulumi-bot/emptygo/p-it-appveyor-v-go-119bcb5f/updates/1
         [                       empty\go ] 
         [                       empty\go ] Wrote output to empty\go\command-output\pulumi-update-initial.20200127-170806.53f1c.log
         [                       empty\go ] Destroying stack
         [                       empty\go ] **** Invoke 'C:\Pulumi\bin\pulumi.exe destroy --non-interactive --skip-preview' in 'empty\go'
         [                       empty\go ] Wrote output to empty\go\command-output\pulumi-destroy.20200127-170809.eeda1.log
         [                       empty\go ] **** Invoke 'C:\Pulumi\bin\pulumi.exe stack rm --yes' in 'empty\go'
         [                single_resource ] Wrote output to C:\Users\appveyor\AppData\Local\Temp\1\p-it-appveyor-v-single_res-95c9f6be-584558113\command-output\pulumi-update-initial.20200127-170810.855a7.log
```

Thoughts on this?

Comment (stack72) [reply tests/integration/integration_test.go]: FWIW, this is the only empty go test that has this...
